srtcore/core.cpp

```diff
@@ -7941,47 +7782,24 @@ int CUDT::packLostData(CPacket &packet, uint64_t &origintime)
     return 0;
 }
 
-int CUDT::packData(CPacket &packet, uint64_t &ts_tk)
+// ts_tk = 0; does not make sense, because return value is checked.
+// Both 0 and -1 will result in no rescheduling.
+std::pair<int, steady_clock::time_point> CUDT::packData(CPacket &packet)
 {
-    int      payload           = 0;
-    bool     probe             = false;
-    uint64_t origintime        = 0;
-    bool     new_packet_packed = false;
-    bool     filter_ctl_pkt    = false;
+   int payload = 0;
+   bool probe = false;
+   steady_clock::time_point origintime;
+   bool new_packet_packed = false;
+   bool filter_ctl_pkt = false;
 
     int kflg = EK_NOENC;
 
-    uint64_t entertime_tk;
-    CTimer::rdtsc(entertime_tk);
+    const steady_clock::time_point enter_time = steady_clock::now();
 
-#if 0 // debug: TimeDiff histogram
-   static int lldiffhisto[23] = {0};
-   static int llnodiff = 0;
-   if (m_ullTargetTime_tk != 0)
-   {
-      int ofs = 11 + ((entertime_tk - m_ullTargetTime_tk)/(int64_t)m_ullCPUFrequency)/1000;
-      if (ofs < 0) ofs = 0;
-      else if (ofs > 22) ofs = 22;
-      lldiffhisto[ofs]++;
-   }
-   else if(m_ullTargetTime_tk == 0)
-   {
-      llnodiff++;
-   }
-   static int callcnt = 0;
-   if (!(callcnt++ % 5000)) {
-      fprintf(stderr, "%6d %6d %6d %6d %6d %6d %6d %6d %6d %6d %6d %6d\n",
-        lldiffhisto[0],lldiffhisto[1],lldiffhisto[2],lldiffhisto[3],lldiffhisto[4],lldiffhisto[5],
-        lldiffhisto[6],lldiffhisto[7],lldiffhisto[8],lldiffhisto[9],lldiffhisto[10],lldiffhisto[11]);
-      fprintf(stderr, "%6d %6d %6d %6d %6d %6d %6d %6d %6d %6d %6d %6d\n",
-        lldiffhisto[12],lldiffhisto[13],lldiffhisto[14],lldiffhisto[15],lldiffhisto[16],lldiffhisto[17],
-        lldiffhisto[18],lldiffhisto[19],lldiffhisto[20],lldiffhisto[21],lldiffhisto[21],llnodiff);
-   }
-#endif
-    if ((0 != m_ullTargetTime_tk) && (entertime_tk > m_ullTargetTime_tk))
-        m_ullTimeDiff_tk += entertime_tk - m_ullTargetTime_tk;
+    if (!is_zero(m_nextSendTime))
```

Comment:
The second condition in the original if() statement is dropped without explanation or obvious rational. If the drop is intentional, is it possible to add a short description of the intended behavior? This may help months from now when we will be looking at this change and wondering why this was done. Thanks!